### PR TITLE
Fix a hang in ResonanceMolSupplier when the computed ceMap is empty

### LINKED ISSUE #5406: Chem.ResonanceMolSupplier freezes/hangs for CC=[N+]=[N-]
**Describe the bug**
Chem.ResonanceMolSupplier hangs (enters an infinite loop?) when applied to `CC=[N+]=[N-]` 

**To Reproduce**
```python
from rdkit import Chem
mol = Chem.MolFromSmiles('CC=[N+]=[N-]')
mol = Chem.AddHs(mol)
supp = Chem.ResonanceMolSupplier(mol)
supp[0]
```
**Expected behavior**
I'm not a Chemist so I'm not sure If there should be more than one resonance structure. However from a code behavior perspective I would at least expect it to not hang and possibly throw an error. 

**Configuration (please complete the following information):**
 - RDKit version: '2022.03.1'
 - OS: macOS 11.4 and Ubuntu 20.4
 - Python version (if relevant): 3.9
 - Are you using conda? yes
 - If you are using conda, which channel did you install the rdkit from? conda-forge
 - If you are not using conda: how did you install the RDKit?

### PR DESCRIPTION
**Pull request by bp-kelley**
Fixes #5406 

Simple fix to prevent entering an infinite loop.

### CODE REVIEW

**Comment by bp-kelley** [comment]
@ptosco  I think this fix is correct.

File: Code/GraphMol/Resonance.cpp [hunk 1]
```diff
@@ -834,6 +834,7 @@ bool ConjElectrons::purgeMaps(CEMap &ceMap, CEDegCount &ceDegCount,
   bool ok = true;
   bool changed = true;
   while (changed) {
+    changed = false;
     for (auto it = ceMap.begin(); it != ceMap.end();) {
       if (!it->second->checkMetrics(ceStats, changed)) {
```
Comment (bp-kelley) [inline Code/GraphMol/Resonance.cpp:839]: checkMetrics changes the value of changed, but if ceMap is empty, this never gets called and changed remains true causing an infinite loop.

Comment (ptosco) [reply Code/GraphMol/Resonance.cpp:839]: Yes, of course, you are right. My bad - I should have engaged my brain before replying :/.

File: Code/GraphMol/resMolSupplierTest.cpp [hunk 2]
```diff
@@ -960,6 +961,19 @@ void testGitHub3349() {
...
+    BOOST_LOG(rdInfoLog) << "-----------------------\n"
+                         << "testGitHub5406" << std::endl;
+    RWMol *mol;
+    mol = SmilesToMol("CC=[N+]=[N-]");
+    MolOps::addHs(*mol);
+    auto *suppl = new ResonanceMolSupplier((ROMol &)*mol);
+    auto length = suppl->length();
+    TEST_ASSERT(length == 0);
+    delete mol;
+    delete suppl;
```
Comment (ptosco) [inline Code/GraphMol/resMolSupplierTest.cpp:974]:
```suggestion
    std::vector<ROMOL_SPTR> mols{"CC=[N+]=[N-]"_smiles, "O=[N+][O-]"_smiles};
    for (auto mol : mols) {
      ResonanceMolSupplier suppl(*mol);
      TEST_ASSERT(suppl.length() == 0);
    }
```

Comment (ptosco) [reply Code/GraphMol/resMolSupplierTest.cpp:974]: Small suggested change to make the test a bit simpler and also add a further test case from #4884, which is indeed caused by the same underlying issue.

**Review by ptosco — APPROVED** [review]
Thanks Brian for finding this simple fix, and sorry being sloppy upon first reading your comment!

File: Code/GraphMol/resMolSupplierTest.cpp (hunk 2 above)
Comment (greglandrum) [reply Code/GraphMol/resMolSupplierTest.cpp:974]: @bp-kelley : do you plan to add these changes too? I think it'd help to have the tests for #4884 there too.

File: Code/GraphMol/resMolSupplierTest.cpp [hunk 3]
```diff
@@ -960,6 +961,19 @@ void testGitHub3349() {
   delete mol;
 }
 
+void testGitHub5406() {
+    BOOST_LOG(rdInfoLog) << "-----------------------\n"
+                         << "testGitHub5406 and 4884" << std::endl;
+    for (auto smiles_string : {"CC=[N+]=[N-]", "O=[N+][O-]"}) {
```
Comment (bp-kelley) [inline Code/GraphMol/resMolSupplierTest.cpp:967]: Can I just say how cool it is that we can do this now?

**Comment by bp-kelley** [comment]
@greglandrum @ptosco I think this is ready for merge now.

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**